frontend/src/components/WalletMenu.vue

```diff
@@ -18,33 +21,48 @@
 </template>
 <script setup lang="ts">
 import { storeToRefs } from 'pinia';
-import { watch } from 'vue';
+import { onMounted, ref, watch } from 'vue';
 
 import { useWallet } from '@/composables/useWallet';
 import { useConfiguration } from '@/stores/configuration';
 import { useEthereumProvider } from '@/stores/ethereum-provider';
 import { useSettings } from '@/stores/settings';
+import { isMobile } from '@/utils/userAgent';
 
 const { provider, signer } = storeToRefs(useEthereumProvider());
 const { rpcUrls } = storeToRefs(useConfiguration());
 const { connectedWallet } = storeToRefs(useSettings());
 const { connectMetaMask, connectWalletConnect } = useWallet(provider, connectedWallet, rpcUrls);
 
-const walletOptions = [
+const walletOptions = ref([
   {
     name: 'MetaMask',
     icon: new URL('../assets/images/metamask.svg', import.meta.url).href,
     description: 'Connect using browser wallet',
-    connect: connectMetaMask,
+    connect: () => connectMetaMask(true),
   },
   {
     name: 'WalletConnect',
     icon: new URL('../assets/images/walletconnect.svg', import.meta.url).href,
     description: 'Connect using mobile wallet',
     connect: connectWalletConnect,
   },
-];
+]);
 
+onMounted(() => {
+  // Use stripped down option list for mobile devices
+  if (isMobile(window.navigator.userAgent)) {
+    const metaMaskAvailable = window.ethereum && window.ethereum.isMetaMask;
```

Comment:
I tried using this function but unfortunately this function took ~5 seconds to execute when accessing via mobile or other device that has no MetaMask. Possibly due to the default timeout set by the library. 
In this case, we cannot afford to let the user wait until we load all the options since the option list is determined based on the providers supported by the device.
This was the reason why i chose to stick with checking on `window.ethereum && window.ethereum.isMetaMask`. 
You can give it a try locally and see the result.

Reply:
Hm ok I'm fine with it. According to [this](https://github.com/MetaMask/detect-provider#synchronous-and-asynchronous-injection) the mobile metamask provider is injected asynchronously. So this works only because the component is not mounted on page load.

Reply:
Ah I didn't see your change earlier. I think it should already be injected when this component is mounted. Reducing the timeout only adds additional logic which I don't think is necessary here. I would leave it as it is.

Reply:
Sorry i didn't notify you on the comment update.
I tried the "refactored" approach but came to a point where i realised we are just introducing more unnecessary logic with it, as you said. 
The only real way to improve this would be to refactor the `useWallet` composable & `WalletMenu` component by having the logic for filtering? and detecting if the providers exists as well as the wallet provider options inside the composable itself.
I am also not fully satisfied with the current solution but i will anyways prioritize writing tests in the next sprints. That will mean that we need to refactor the components to align with the tests, therefore we will address this topic better in the following weeks.

Thanks for the input!